curated_transformers/models/hf_wrapper.py

```diff
@@ -72,24 +70,30 @@ def convert_from_torch_backward(d_inputs):
 
 
 def _convert_outputs(model, inputs_outputs, is_train):
-    X, model_outputs = inputs_outputs
+    ops = get_current_ops()
 
     # Strip the outputs for the padding timesteps
-    # and return the outputs from the last Transformer layer.
-    last_layer_output = model_outputs.last_hidden_state
-
-    X_lens = [x.shape[0] for x in X]
-    Yt = [last_layer_output[i, :len, :] for i, len in enumerate(X_lens)]
-    Y = [torch2xp(yt) for yt in Yt]
+    # while preserving all the outputs (and their order).
+    model_inputs, model_outputs = inputs_outputs
+    input_lens = [x.shape[0] for x in model_inputs]
+    Yt = [
+        [output[i, :len, :] for output in model_outputs.all_outputs]
+        for i, len in enumerate(input_lens)
+    ]
+    Y = [[torch2xp(layer, ops=ops) for layer in output] for output in Yt]
+    output = TransformerModelOutput(outputs=Y)
+
+    def convert_for_torch_backward(dY: List[List[Floats2d]]):
+        Yt_flat = [y for inner in Yt for y in inner]
+        dYt_flat = [xp2torch(y) for inner in dY for y in inner]
```

Comment:
I think this already came up in an earlier discussion -- we'd always backprop through all layers, even if we only use last layer listeners. The performance impact may not be that bad, since the gradients are accumulated, but did you have a chance to measure the difference?

Reply:
Not yet, but I can train a model to check before we merge this PR.

Reply:
The performance impact of processing all layer outputs is significant (~20-23% reduction in WPS), but the latest changes should restrict the penalty to only those cases where all outputs are used downstream. When they aren't, we simply remove all but the last hidden representation from the `TransformerOutput` instance, thereby avoiding redundant processing.